trie/triedb/hashdb/database.go

```diff
@@ -65,6 +65,17 @@ type ChildResolver interface {
 	ForEach(node []byte, onChild func(common.Hash))
 }
 
+// Config contains the settings for database.
+type Config struct {
+	CleanSize int // Maximum memory allowance (in bytes) for caching clean nodes
+}
+
+// Defaults is the default setting for database if it's not specified.
+// Notably, clean cache is disabled explicitly, since Reset operation
```

Comment:
Why do you mention that Reset must be called in the config? Isn't it done internally by the database as long as it's explicitly Close-d? If the latter, perhaps just mention that the caller needs to Close the database instead of referencing Reset, which is an internal impl detail.